node/src/asset_cleanup.rs

```diff
@@ -0,0 +1,416 @@
+use crate::assets::clean_db;
+use crate::config::ParticipantsConfig;
+use crate::db::{DBCol, SecretDB, EPOCH_ID_KEY};
+use crate::primitives;
+use crate::providers::ecdsa::presign::PresignOutputWithParticipants;
+use crate::providers::ecdsa::triple::{PairedTriple, TRIPLE_STORE_DOMAIN_ID};
+use mpc_contract::primitives::domain::DomainId;
+use mpc_contract::primitives::key_state::EpochId;
+use serde::{self, Deserialize, Serialize};
+use std::sync::Arc;
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct EpochData {
+    pub epoch_id: EpochId,
+    pub participants: ParticipantsConfig,
+}
+
+/* database helpers */
+enum EpochDataWrapper {
+    Legacy(EpochId),
+    Current(EpochData),
+}
+
+fn get_epoch_data(db: &Arc<SecretDB>) -> anyhow::Result<Option<EpochDataWrapper>> {
+    let Some(db_res): Option<Vec<u8>> = db.get(DBCol::EpochData, EPOCH_ID_KEY)? else {
+        return Ok(None);
+    };
+    if let Ok(epoch_data) = bincode::deserialize::<EpochData>(&db_res) {
+        return Ok(Some(EpochDataWrapper::Current(epoch_data)));
+    }
+
+    if db_res.len() == 8 {
+        let bytes_array: [u8; 8] = db_res
+            .as_slice()
+            .try_into()
+            .inspect_err(|bytes| tracing::error!("PREVIOUS EPOCH_ID ENTRY NOT u64: {:?}", bytes))?;
+        let epoch_id_number = u64::from_be_bytes(bytes_array);
+        let epoch_id = EpochId::new(epoch_id_number);
+        return Ok(Some(EpochDataWrapper::Legacy(epoch_id)));
+    };
+    anyhow::bail!("Can't deserialize EPOCH_ID entry: {:?}", db_res);
+}
+
+enum AssetCleanup {
+    Keep,
+    DeleteAll,
+    KeepOnly(Vec<primitives::ParticipantId>),
+}
+
+fn get_cleanup_behavior(
```

Comment:
Nit: I find the `get_` prefix a bit confusing here since we're not getting anything from a database or external service.

```suggestion
fn cleanup_behavior(
```